Sources/Datadog/FeaturesIntegration/CrashReporting/CrashReportingWithRUMIntegration.swift

```diff
@@ -53,34 +66,150 @@ internal struct CrashReportingWithRUMIntegration: CrashReportingIntegration {
-        // approximation we can get.
+        // to the server time before processing. Following use of the current correction is not ideal (it's not the correction
+        // from the moment of crash), but this is the best approximation we can get.
         let currentTimeCorrection = dateCorrector.currentCorrection
 
         let crashDate = crashReport.date ?? dateProvider.currentDate()
-        let realCrashDate = currentTimeCorrection.applying(to: crashDate)
-        let realDateNow = currentTimeCorrection.applying(to: dateProvider.currentDate())
+        let adjustedCrashTimings = AdjustedCrashTimings(
+            crashDate: crashDate,
+            realCrashDate: currentTimeCorrection.applying(to: crashDate),
+            realDateNow: currentTimeCorrection.applying(to: dateProvider.currentDate())
+        )
 
-        if realDateNow.timeIntervalSince(realCrashDate) < Constants.viewEventAvailabilityThreshold {
-            let rumError = createRUMError(from: crashReport, and: lastRUMViewEvent, crashDate: realCrashDate)
-            let rumView = updateRUMViewWithNewError(lastRUMViewEvent, crashDate: realCrashDate)
-            rumEventOutput.write(rumEvent: rumError)
-            rumEventOutput.write(rumEvent: rumView)
+        if let lastRUMViewEvent = crashContext.lastRUMViewEvent {
+            sendCrashReportLinkedToLastViewInPreviousSession(crashReport, lastRUMViewEventInPreviousSession: lastRUMViewEvent, using: adjustedCrashTimings)
+        } else if let lastRUMSessionState = crashContext.lastRUMSessionState {
+            sendCrashReportToPreviousSession(crashReport, crashContext: crashContext, lastRUMSessionStateInPreviousSession: lastRUMSessionState, using: adjustedCrashTimings)
+        } else if rumConfiguration.sessionSampler.sample() { // before producing a new RUM session, we must consider sampling
+            sendCrashReportToNewSession(crashReport, crashContext: crashContext, using: adjustedCrashTimings)
         } else {
-            let rumError = createRUMError(from: crashReport, and: lastRUMViewEvent, crashDate: realCrashDate)
+            userLogger.info("There was a crash in previous session, but it is ignored due to sampling.")
+        }
+    }
+
+    /// If the crash occured in an existing RUM session and we know its `lastRUMViewEvent` we send the error using that session UUID and link
+    /// the crash to that view. The error event can be preceded with a view update based on `Constants.viewEventAvailabilityThreshold` condition.
+    private func sendCrashReportLinkedToLastViewInPreviousSession(
+        _ crashReport: DDCrashReport,
+        lastRUMViewEventInPreviousSession lastRUMViewEvent: RUMEvent<RUMViewEvent>,
+        using crashTimings: AdjustedCrashTimings
+    ) {
+        if crashTimings.realDateNow.timeIntervalSince(crashTimings.realCrashDate) < Constants.viewEventAvailabilityThreshold {
```

Comment:
Yes, it's been like this since beginning of crash reporting. Problem is that backend considers all RUM sessions as "closed" after `4h`, so sending additional view update to it might introduce inconsistency. It was discussed with BE team, and this is the best we came up with.

Reply:
Ok 👍 Shouldn't be a big deal in that specific case and totally out of the scope of this PR, but I'm concerned that if we change that value BE side, for any reason, we will have inconsistencies.
Would be great to pull some configuration from BE at some point.